### PR TITLE
test(dashboard): add unit tests for AI helper functions

### PR DESCRIPTION
**Pull request by abhigyan1102**
## Summary

Adds comprehensive unit tests for `packages/dashboard/src/features/ai/helpers.ts` — the dashboard module containing all AI model formatting, filtering, and display logic. This file (244 lines, 15+ exported functions) previously had **zero test coverage**.

## Motivation

These helper functions are used across the AI Models page, the model selector, and the overview dashboard to:
- Filter models by provider and modality
- Generate dynamic provider tabs (OpenAI, Anthropic, Google, etc.)
- Format prices, token counts, dates, and modality labels
- Map raw API model schemas to display-ready options

Without tests, any refactor to the filtering or formatting logic risks breaking the entire AI dashboard UI silently.

### Provider Extraction & Display
- `getProviderIdFromModelId` — extracts provider from `openai/gpt-5.5` format
- `getProviderDisplayName` — maps IDs to display names (openai→OpenAI, x-ai→xAI), case-insensitive, unknown provider capitalization fallback
- `getProviderDisplayOrder` — sort priority for known providers, fallback for unknown

### Model Filtering
- `filterModelsByProvider` — exact provider match, "other" tab aggregation for providers without logos
- `filterModelsByModalities` — input/output modality filtering, handles empty model lists and null safety

### Tab Generation
- `generateProviderTabs` — creates tabs for providers with logos, adds "Other" tab for unknown providers, sorts by display order, handles empty model list

### Formatting
- `formatTokenCount` — M/K/raw thresholds (1.5M, 128.0K, 999)
- `getFriendlyModelName` — kebab-case to Title Case conversion
- `formatCredits` — K suffix for thousands, 2 decimal places for small values
- `formatPrice` — undefined→"-", zero→"Free", small→4 decimals, medium→2 decimals, large→1 decimal
- `formatInputPrice`/`formatOutputPrice` — prefers server-provided label, falls back to formatPrice
- `formatModality` — first-letter capitalization
- `formatReleasedDate` — Unix timestamp to MM/DD/YYYY, null/undefined→"-"

### Schema Mapping
- `toModelOption` — maps AIModelSchema to ModelOption with provider name, friendly model name, and logo

## How I tested this

```bash
npx vitest run --config vitest.unit.config.ts src/features/ai/__tests__/helpers.test.ts  # 46 passed
npx vitest run --config vitest.unit.config.ts                                             # Full dashboard suite passed

<!-- This is an auto-generated description by cubic. -->
---
## Summary by cubic
Add unit tests for `packages/dashboard/src/features/ai/helpers.ts`, covering provider parsing, model filtering, tab generation, formatting, and schema mapping. Also replace `as any` with typed casts, remove a duplicate `modelId` in the test factory to fix TS2783, and run Prettier; 46 tests to harden the Models page, model selector, and overview.

<sup>Written for commit 40fe418d30f2295dd30a941e9ad7a2e98e1c6770. Summary will update on new commits. <a href="https://cubic.dev/pr/InsForge/InsForge/pull/1316?utm_source=github">Review in cubic</a></sup>

<!-- End of auto-generated description by cubic. -->

<!-- Macroscope's pull request summary starts here -->
<!-- Macroscope will only edit the content between these invisible markers, and the markers themselves will not be visible in the GitHub rendered markdown. -->
<!-- If you delete either of the start / end markers from your PR's description, Macroscope will append its summary at the bottom of the description. -->
> [!NOTE]
> ### Add unit tests for AI dashboard helper functions
> Adds a new [helpers.test.ts](https://github.com/InsForge/InsForge/pull/1316/files#diff-815f6b3ca424fdeb4e3a4d997f7f2f0d88a4283de85badd758736ae3d0b64fc6) vitest suite covering 14 helper functions including provider lookup, model filtering, tab generation, and price/date formatting. Tests cover edge cases such as case-insensitivity, fallback behavior, 'other' tab logic, and modality filtering.
>
> <!-- Macroscope's review summary starts here -->
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized 40fe418.</sup>
> <!-- Macroscope's review summary ends here -->
>
<!-- macroscope-ui-refresh -->
<!-- Macroscope's pull request summary ends here -->

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Tests**
  * Added comprehensive unit tests for AI helper utilities.
  * Covers provider ID extraction and provider display/sort mapping (including case-insensitivity and unknown providers).
  * Verifies model filtering by provider and modality, provider tab generation with “other” grouping, and deterministic ordering.
  * Validates formatting helpers for tokens, friendly model names, credits/prices, modality capitalization, and release date handling.

<!-- review_stack_entry_start -->

[![Review Change Stack](https://storage.googleapis.com/coderabbit_public_assets/review-stack-in-coderabbit-ui.svg)](https://app.coderabbit.ai/change-stack/InsForge/InsForge/pull/1316?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

<!-- review_stack_entry_end -->
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

### CODE REVIEW

File: packages/dashboard/src/features/ai/__tests__/helpers.test.ts [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+describe('getProviderIdFromModelId', () => {
+  it('extracts provider from standard model ID', () => {
+    expect(getProviderIdFromModelId('openai/gpt-5.5')).toBe('openai');
+  });
+
+  it('extracts provider from multi-segment model ID', () => {
+    expect(getProviderIdFromModelId('google/gemini-2.5-pro')).toBe('google');
+  });
+
+  it('returns empty string for model ID without slash', () => {
```
Comment (Fermionic-Lyu) [inline packages/dashboard/src/features/ai/__tests__/helpers.test.ts:41]: This test name says the helper returns an empty string for a model ID without a slash, but the assertion expects `standalone`. Either the expected behavior or the description should change so the next reader does not inherit the ambiguity.

File: packages/dashboard/src/features/ai/__tests__/helpers.test.ts [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+  it('handles empty string', () => {
+    expect(formatModality('')).toBe('');
+  });
+});
+
+describe('formatReleasedDate', () => {
+  it('formats Unix timestamp to MM/DD/YYYY', () => {
+    // 1700000000 = Nov 14, 2023
+    const result = formatReleasedDate(1700000000);
+    expect(result).toMatch(/^\d{2}\/\d{2}\/\d{4}$/);
```
Comment (Fermionic-Lyu) [inline packages/dashboard/src/features/ai/__tests__/helpers.test.ts:317]: This only asserts the output has an `MM/DD/YYYY` shape, so a regression that formats the Unix seconds value as milliseconds would still pass with a 1970 date. Please assert the actual expected date here, or otherwise make the seconds-to-milliseconds conversion observable.
